vector/src/main/java/im/vector/app/features/roomdirectory/createroom/CreateRoomFragment.kt

```diff
@@ -163,8 +163,9 @@ class CreateRoomFragment @Inject constructor(
     }
 
     override fun selectVisibility() = withState(viewModel) { state ->
-
-        val allowed = if (state.supportsRestricted) {
+        // If restricted is supported and the user is in the context of a parent space
+        // then show restricted option.
+        val allowed = if (state.supportsRestricted && state.parentSpaceId != null) {
```

Comment:
NIT: if we wanted to avoid using separate branches for appending to the list (and potentially be more kotlin idiomatic) we could use 

`listOfNotNull`

```kotlin
val list = listOfNotNull(
        INVITE,
        PUBLIC,
        RESTRICTED.takeIf { state.supportsRestricted && state.parentSpaceId != null }
)
```

or `buildList`

```kotlin
val list = buildList {
    add(INVITE)
    add(PUBLIC)
    if (state.supportsRestricted && state.parentSpaceId != null) {
        add(RESTRICTED)
    }
}
```

Reply:
AS is telling me that it's experimental

Reply:
ah my bad, I totally forgot that `buildList` is still experimental